### PR TITLE
chore: Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

**What is the previous behavior before this PR?**

The misspellings have been reported at https://github.com/jsoref/KaTeX/commit/55adbfb034441663fe62209bd9442cd750a7869b

**What is the new behavior after this PR?**

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/KaTeX/commit/dfc30bb80e04e50d98d0ab4347acebe15bfc227c

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -352,7 +352,7 @@ You can now use:
 - docs: Update \operatorname in supported_table.md 0.12.0 (#2571)
 - docs: Fix documentation typo in operatorname* (#2570)
 - docs: add warning re:defer to mhchem documentation (#2485)
-- docs: update Gastby logo and link (#2481)
+- docs: update Gatsby logo and link (#2481)
```
Comment (jsoref) [inline CHANGELOG.md:355]: This is a brand.

File: contrib/mhchem/mhchem.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ﻿/* eslint-disable */
-/* -*- Mode: Javascript; indent-tabs-mode:nil; js-indent-level: 2 -*- */
+/* -*- Mode: JavaScript; indent-tabs-mode:nil; js-indent-level: 2 -*- */
```
Comment (jsoref) [inline contrib/mhchem/mhchem.js:2]: Brand

File: src/delimiter.js [hunk 3]
```diff
@@ -453,7 +453,7 @@ const makeStackedDelim = function(
         Style.TEXT, options, classes);
 };
 
-// All surds have 0.08em padding above the viniculum inside the SVG.
+// All surds have 0.08em padding above the vinculum inside the SVG.
```
Comment (jsoref) [inline src/delimiter.js:456]: This item is consistently spelled this way, but the world disagrees...

https://www.merriam-webster.com/dictionary/vinculum

this is a TI(re)L - Google Sheets told me.

File: src/environments/array.js [hunk 4]
```diff
@@ -573,8 +573,8 @@ const mathmlBuilder: MathMLBuilder<"array"> = function(group, options) {
     // We simulate this by adding (arraystretch - 1)em to the gap. This
     // does a reasonable job of adjusting arrays containing 1 em tall content.
 
-    // The 0.16 and 0.09 values are found emprically. They produce an array
-    // similar to LaTeX and in which content does not interfere with \hines.
+    // The 0.16 and 0.09 values are found empirically. They produce an array
+    // similar to LaTeX and in which content does not interfere with \hlines.
```
Comment (jsoref) [inline src/environments/array.js:577]: I'm not sure what `\hines` is. I'm _guessing_ it's a typo. But I could be very wrong.

**Review by jsoref — COMMENTED** [review]
Most corrections automatically suggested by Google Sheets. All fault mine.

File: src/environments/array.js (hunk 4 above)
Comment (ronkok) [reply src/environments/array.js:577]: It's a typo.

**Comment by ronkok** [comment]
Good stuff.

@edemaine I think you are the right person to decide if the variable name in `macros.js` should change from `nonexistsOK` to `nonexistentOK`. There might be some small nuance in play there.

All the rest looks good to me.

**Comment by edemaine** [comment]
Thanks @ronkok for reviewing! I agree that `nonexistsOK` would be better, for parallelism with `existsOK`.  What we mean by `nonexistsOK` is "is it OK that the command doesn't exist?"  Another possible spelling would be `notExistsOK`... is that better/clearer?

**Comment by edemaine** [comment]
P.S. I think adding the "check spelling" action would be cool, as a separate PR, if there aren't any false positives (or there's an easy way to deal with them).

**Comment by jsoref** [comment]
Oops. Sorry (I failed to properly fill out the description -- fixed...). In regard to adding check-spelling..., you can see the configuration for check-spelling here: https://github.com/jsoref/KaTeX/commit/55adbfb034441663fe62209bd9442cd750a7869b


functionally, https://github.com/jsoref/KaTeX/blob/55adbfb034441663fe62209bd9442cd750a7869b/.github/actions/spelling/expect.txt is a list of terms that check-spelling doesn't naturally recognize (with the addition of [supplemental dictionaries](https://github.com/jsoref/KaTeX/blob/55adbfb034441663fe62209bd9442cd750a7869b/.github/workflows/spelling.yml#L87-L98)) and that I didn't recognize as typos of anything else. You could think of them as "false-positives".

As for dealing w/ false-positives, I've been working on improving the flow. At this point, the current "easiest" approach can be seen in https://github.com/jsoref/KaTeX/commit/55adbfb034441663fe62209bd9442cd750a7869b#commitcomment-94184527

> To accept ✔️ these unrecognized words as correct and remove the previously acknowledged and now absent words, run the following commands

Basically the command will update the metadata and let you commit them. There's also a provision for updating metadata using a PR comment (which my company uses), but I don't have examples of it being generally deployed in public repositories.

I welcome feedback (it does take me some time to digest the feedback and implement a response, and this is a side project, but I try to be fairly responsive).

**Review by edemaine — APPROVED** [review]
Sorry for the delay on this. Feel free to submit a PR that adds this to the CI workflow; I imagine you'd be the expert on that.
